src/client/state/sagas/index.js

```diff
@@ -0,0 +1,182 @@
+import {
+  LOAD_HOMEPAGE,
+  DISPOSE_HOMEPAGE,
+  FETCH_ARTICLES_BY_TOPIC,
+  LOAD_ARTICLE_PAGE,
+  DISPOSE_ARTICLE_PAGE,
+  LOAD_ARCHIVES_PAGE,
+  DISPOSE_ARCHIVES_PAGE,
+  FETCH_ARCHIVE_BY_ID,
+  LOAD_USERS_PAGE,
+  DISPOSE_USERS_PAGE,
+  LOAD_EDIT_USER,
+  DISPOSE_EDIT_USER,
+  LOAD_TOPICS_PAGE,
+  DISPOSE_TOPICS_PAGE,
+  LOAD_EDIT_TOPIC,
+  DISPOSE_EDIT_TOPIC
+} from "state/actions/types";
+import { call, put, takeEvery, takeLatest } from "redux-saga/effects";
+
+import {
+  addArticles,
+  emptyArticles,
+  startLoadingArticles,
+  stopLoadingArticles,
+  setCurrentArticle,
+  emptyCurrentArticle
+} from "state/actions/article";
+
+import {
+  addArchives,
+  startLoadingArchives,
+  stopLoadingArchives,
+  emptyArchives,
+  emptyCurrentArchive,
+  setCurrentArchive
+} from "state/actions/archive";
+
+import {
+  addUsers,
+  emptyUsers,
+  startLoadingUsers,
+  stopLoadingUsers,
+  setCurrentUser,
+  emptyCurrentUser
+} from "state/actions/user";
+
+import {
+  addTopics,
+  emptyTopics,
+  startLoadingTopics,
+  stopLoadingTopics,
+  setCurrentTopic,
+  emptyCurrentTopic
+} from "state/actions/topic";
+
+import { startLoadingApp, stopLoadingApp } from "state/actions/app";
+
+import APIProvider from "utils/APIProvider";
+
+function* loadHomepage() {
+  yield put(startLoadingApp());
+  const topics = yield call(APIProvider.get, "topics");
+  const articles = yield call(APIProvider.get, "articles");
+  yield put(addTopics(topics));
+  yield put(addArticles(articles));
+  yield put(stopLoadingApp());
+}
+
+function* disposeHomepage() {
+  yield put(emptyArticles());
+  yield put(emptyTopics());
+}
+
+function* fetchArticlesByTopic(action) {
+  yield put(startLoadingArticles());
+  const topic = yield call(APIProvider.get, `topics/${action.id}/articles`);
```

Comment:
After the stuff in `model-updates-bugs` PR is merged, this could become:

```
 const topic = yield call(APIProvider.get, `articles?topic_id=${action.id}`);
```